dm/syncer/syncer.go

```diff
@@ -756,57 +757,70 @@ func (s *Syncer) getTableInfo(tctx *tcontext.Context, sourceTable, targetTable *
 func (s *Syncer) trackTableInfoFromDownstream(tctx *tcontext.Context, sourceTable, targetTable *filter.Table) error {
 	// TODO: Switch to use the HTTP interface to retrieve the TableInfo directly if HTTP port is available
 	// use parser for downstream.
-	parser2, err := utils.GetParserForConn(tctx.Ctx, s.ddlDBConn.BaseConn.DBConn)
-	if err != nil {
-		return terror.ErrSchemaTrackerCannotParseDownstreamTable.Delegate(err, targetTable, sourceTable)
+	retryStrategy := retry.FiniteRetryStrategy{}
+	params := retry.Params{
+		RetryCount:         10,
+		FirstRetryDuration: retryTimeout,
+		BackoffStrategy:    retry.Stable,
+		IsRetryableFn: func(_ int, _ error) bool {
+			// retry until reaching limit
+			return true
+		},
 	}
+	_, _, err := retryStrategy.Apply(tctx, params, func(tctx *tcontext.Context) (interface{}, error) {
+		parser2, err := utils.GetParserForConn(tctx.Ctx, s.ddlDBConn.BaseConn.DBConn)
```

Comment:
i found tracker has its own conn which set `mysql.DefaultSQLMode` , this may cause the sqlmode to be inconsistent in two places ? see [here](https://github.com/pingcap/tiflow/blob/a9d5d5a6d9120900fdf49ba670cdbaf0b82c04fb/dm/syncer/syncer.go#L3405) and [here](https://github.com/pingcap/tiflow/blob/a9d5d5a6d9120900fdf49ba670cdbaf0b82c04fb/dm/pkg/schema/tracker.go#L543)
cc @WizardXiao

Reply:
set `mysql.DefaultSQLMode` is to advoid downstream sql_mode changing and we can not parse the create sql, so we set a high compatibility sql_mode.

Reply:
if so can we just use `mysql.DefaultSQLMode` here ?

Reply:
if ok to use mysql.DefaultSQLMode(for compatibility), no need retry anymore
 cc @buchuitoudegou @lance6716

Reply:
> no need retry anymore

please read the issue to know what problem retry resolves

Reply:
> if ok to use mysql.DefaultSQLMode(for compatibility), no need retry anymore cc @buchuitoudegou @lance6716

This PR attempts to make tracker automatically retry for SQL querying when querying fails due to some network issue.

Reply:
ohoho mybad, i  for GetTableCreateSQL,I missed this `GetTableCreateSQL ` before, I thought you just want add retry for `GetParserForConn `